integration-tests/features/steps/component_analysis_v2.py

```diff
@@ -0,0 +1,163 @@
+"""Tests for API endpoints that performs v2 component analysis."""
+import requests
+
+import time
+
+from behave import then, when
+from urllib.parse import urljoin
+
+from src.parsing import parse_token_clause
+
+
+def component_analysis_v2_url(context, ecosystem, component, version):
+    """Construct URL for the component analyses REST API call."""
+    return urljoin(context.core_v2_api_url,
+                   '/api/v2/component-analyses/{e}/{c}/{v}'.format(e=ecosystem,
+                                                                   c=component,
+                                                                   v=version))
+
+
+def perform_v2_component_analysis(context, ecosystem, package, version, use_user_key, rate):
+    """Call API endpoint to v2 analysis for component."""
+    context.duration = None
+    start_time = time.time()
+    url = component_analysis_v2_url(context, ecosystem, package, version)
+    if use_user_key:
+        for _ in range(rate):
+            context.response = requests.get(
+                url, params={'user_key': context.three_scale_preview_user_key})
+            # break the loop if rate limit exceeded
+            if context.response.status_code == 429:
+                break
+    else:
+        context.response = requests.get(url)
```

Comment:
Dont you need to iterate through the rate in this scenario?

Reply:
i did this to add a case since our 3scale accepts 100 reqs per min i will try to do more requests in 1 min and i will see if i get 429 error , if not specified the rate is always 1

Reply:
Ok, but dont we need to test the same for this part also? if the rate is 1, then it will iterate only once.

Reply:
i was about to add a case but when i test locally even if i send 120 requests its working i will see once on ci

Reply:
added the 429 test case now